sigopt/xgboost/run.py

```diff
@@ -0,0 +1,109 @@
+import platform
+import copy
+
+import xgboost
+# pylint: disable=no-name-in-module
+from xgboost import DMatrix
+
+from ..context import Context
+from ..log_capture import SystemOutputStreamMonitor
+from .. import create_run
+
+DEFAULT_EVALS_NAME = 'Test Set'
+DEFAULT_RUN_OPTIONS = {
+  'log_sys_info': True,
+  'log_stdout': True,
+  'log_stderr': True,
+  'run': None
+}
+XGB_ALIASES = \
+  {
+     'learning_rate': 'eta',
+     'min_split_loss': 'gamma',
+     'reg_lambda': 'lambda',
+     'reg_alpha': 'alpha'
+  }
+
+
+def parse_run_options(run_options):
+  if run_options:
+    assert run_options.keys() <= DEFAULT_RUN_OPTIONS.keys(), 'Unsupported argument inside run_options'
+  run_options_parsed = {**DEFAULT_RUN_OPTIONS, **run_options} if run_options else DEFAULT_RUN_OPTIONS
+  return run_options_parsed
+
+
+class XGBRun:
+
+  def __init__(self, params, dtrain, num_boost_round, evals, run_options):
+    self.params = params
+    self.dtrain = dtrain
+    self.num_boost_round = num_boost_round
+    self.evals = evals
+    self.run_options_parsed = parse_run_options(run_options)
+    self.validation_sets = [(evals, DEFAULT_EVALS_NAME)] if isinstance(evals, DMatrix) else evals
+    self.run = None
+
+  def make_run(self):
+    if self.run_options_parsed['run']:
+      self.run = self.run_options_parsed['run']
+    else:
+      self.run = create_run()
+
+  def log_metadata(self):
+    if self.run_options_parsed['log_sys_info']:
+      python_version = platform.python_version()
+      self.run.log_metadata("Python Version", python_version)
+      self.run.log_metadata("XGBoost Version", xgboost.__version__)
+    self.run.log_model("XGBoost")
+    self.run.log_metadata("_IS_XGB", 'True')
+    self.run.log_metadata("Dataset columns", self.dtrain.num_col())
+    self.run.log_metadata("Dataset rows", self.dtrain.num_row())
+    self.run.log_metadata("Objective", self.params['objective'])
```

Comment:
I think we should log `eval_metric` instead of `objective` as metadata. `eval_metric` can be a list of metrics, so it's best to be logged as meta.

Reply:
I'm ok logging both as metadata as well.